das/worker.go

```diff
@@ -0,0 +1,93 @@
+package das
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"go.uber.org/multierr"
+)
+
+type worker struct {
+	lock     sync.Mutex
+	state    workerState
+	resultCh chan<- result
+	sample   func(context.Context, uint64) error
```

Comment:
why does this have to be stored on the worker? You can call `w.run()` and pass in the samplingCoordinaters sampleFn since it's the same for all sampling routines.